src/main/java/kr/co/antoon/oauth/handler/OAuth2AuthenticationSuccessHandler.java

```diff
@@ -0,0 +1,80 @@
+package kr.co.antoon.oauth.handler;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import kr.co.antoon.oauth.application.TokenService;
+import kr.co.antoon.oauth.dto.Token;
+import kr.co.antoon.oauth.dto.UserDto;
+import kr.co.antoon.oauth.dto.UserRequestMapper;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.oauth2.core.user.OAuth2User;
+import org.springframework.security.web.authentication.SimpleUrlAuthenticationSuccessHandler;
+import org.springframework.stereotype.Component;
+
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+@Slf4j
+@RequiredArgsConstructor
+@Component
+public class OAuth2AuthenticationSuccessHandler extends SimpleUrlAuthenticationSuccessHandler {
+    private final TokenService tokenService;
+    private final UserRequestMapper userRequestMapper;
+    private final ObjectMapper objectMapper;
+
+//    @Override
+//    public void onAuthenticationSuccess(HttpServletRequest request, HttpServletResponse response,
+//                                        Authentication authentication) throws IOException {
+//        OAuth2User oAuth2User = (OAuth2User)authentication.getPrincipal();
+//        UserDto userDto = userRequestMapper.toDto(oAuth2User);
+//
+//        //로그인 성공 시
+//        String url = makeRedirectUrl(tokenProvider.createToken(authentication));
+//        Token token = tokenService.generateToken(userDto.getEmail(), "USER");
+//        log.info("{}", token);
+//
+//        writeTokenResponse(response, token);
+//
+//        if (response.isCommitted()) {
+//            logger.debug("응답이 이미 커밋된 상태입니다. " + url + "로 리다이렉트하도록 바꿀 수 없습니다.");
+//            return;
+//        }
+//        getRedirectStrategy().sendRedirect(request, response, url);
+//    }
+//
+//    private String makeRedirectUrl(String token) {
+//        return UriComponentsBuilder.fromUriString("http://localhost:8080/oauth2/redirect")
+//                .queryParam("token", token)
+//                .build().toUriString();
+//    }
+
+    @Override
+    public void onAuthenticationSuccess(HttpServletRequest request, HttpServletResponse response, Authentication authentication)
+            throws IOException, ServletException {
+        OAuth2User oAuth2User = (OAuth2User)authentication.getPrincipal();
+        UserDto userDto = userRequestMapper.toDto(oAuth2User);
+
+        // 최초 로그인이라면 회원가입 처리를 한다.
```

Comment:
```suggestion
        // TODO: 최초 로그인이라면 회원가입 처리를 한다.
```
이 후에 작업이 필요한 경우에 일반 주석과 구분하기 위해서 TODO 혹은 FIXME 같은 걸 주석으로 달아주면 인텔리제이에서 색상으로 표시되서 나중에 작업 할 곳을 찾기 더 편합니다.